src/main/java/com/sap/oss/phosphor/fosstars/data/owasp/experimental/VulnerabilitiesFromOwaspDependencyCheck.java

```diff
@@ -0,0 +1,404 @@
+      throws IOException {
+    Objects.requireNonNull(artifact, "On no! The artifact object cannot be null");
+    if (!artifact.version().isPresent()) {
+      throw new IOException("Oh no! the version is not available.");
+    }
+
+    Optional<OwaspDependencyCheckEntry> owaspDependencyCheckEntry = scan(artifact);
+    if (!owaspDependencyCheckEntry.isPresent()) {
+      values.update(VULNERABILITIES.unknown());
+      return this;
+    }
+
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    for (Dependency dependency : owaspDependencyCheckEntry.get().getDependencies()) {
+      if (dependency.getVulnerabilities() == null) {
+        continue;
+      }
+
+      for (OwaspDependencyCheckVuln owaspDependencyCheckvuln : dependency.getVulnerabilities()) {
+        vulnerabilities.add(from(owaspDependencyCheckvuln));
+      }
+    }
+
+    values.update(VULNERABILITIES.value(vulnerabilities));
+    return this;
+  }
+
+  /**
+   * Returns the supported feature loaded by this data provider.
+   */
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return setOf(VULNERABILITIES);
+  }
+
+  /**
+   * Scan the input jar file and analyze the extracted {@link Dependency}.
+   * 
+   * @param file The jar.
+   */
+  private void analyze(File file) {
+    try {
+      engine.scan(file);
+      engine.analyzeDependencies();
+    } catch (ExceptionCollection e) {
+      exceptionCollection.addException(e);
+    }
+  }
+
+  /**
+   * Process the report generated from analysis of the jar file.
+   * 
+   * @param fileName The name of report.
+   * @return An optional of {@link OwaspDependencyCheckEntry}.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<OwaspDependencyCheckEntry> process(String fileName) throws IOException {
+    Optional<Path> reportPath = createDirectory(REPORT_DIR);
+
+    if (reportPath.isPresent()) {
+      File file = reportPath.get().resolve(String.format("%s.json", fileName)).toFile();
+
+      try {
+        engine.writeReports(fileName, file, REPORT_OUTPUT_FORMAT, exceptionCollection);
+        ObjectMapper mapper = new ObjectMapper();
+        return Optional.ofNullable(mapper.readValue(file, OwaspDependencyCheckEntry.class));
+      } catch (ReportException e) {
+        throw new IOException("Oh no! the report writing failed.", e);
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Scan the {@link MavenArtifact}.
+   * 
+   * @param artifact The {@link MavenArtifact}.
+   * @return An optional of {@link OwaspDependencyCheckEntry}.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<OwaspDependencyCheckEntry> scan(MavenArtifact artifact) throws IOException {
+    Optional<Path> filePath = fetch(artifact);
+
+    if (filePath.isPresent()) {
+      try {
+        analyze(filePath.get().toFile());
+        return process(filePath.get().toFile().getName());
+      } finally {
+        if (engine != null) {
```

Comment:
Currently, it doesn't look right that the provider closes the engine here. If you call the provider twice, then most likely it fails because the engine is closed. The null-check also looks redundant.

I think it would be better to create and close an engine in this method instead of the constructor.

Note that we have try-with-resources.